rust/src/router/pipe_transport.rs

```diff
@@ -173,7 +171,7 @@ impl PipeTransportDump {
                 .map(TransportTraceEventType::from_fbs)
                 .collect(),
             // PipeTransport specific.
-            tuple: TransportTuple::from_fbs(dump.tuple.as_ref()),
+            tuple: TransportTuple::from_fbs(*dump.tuple),
```

Comment:
IIRC Flatbuffers requires allocations anyway, in which case can't we continue using references everywhere?

Reply:
Yes. I don't think it makes a huge difference here, but generally if we do not consume input type, the argument should be a reference.

Reply:
Done.